src/cmds/fs/more.c

```diff
@@ -17,71 +17,62 @@
 #define TAB_SIZE 4
 #define MAX_SCREEN_WIDTH 256
 
-static char buff[MAX_SCREEN_WIDTH], info[MAX_SCREEN_WIDTH];
+static char buff[MAX_SCREEN_WIDTH];
 
 static void screen(FILE *fp) {
-    int cmd = 0, columns, lines, x, y;
-	WINDOW *std;
-	bool endOfFile = false;
+	int cmd = 0, columns, lines, x, y, i;
 
+	WINDOW *std;
 	std = initscr();
 	columns = std->endx - std->begx;
 	if (columns >= MAX_SCREEN_WIDTH - 1) {
 		columns = MAX_SCREEN_WIDTH - 1;
 	}
-	lines	= std->endy - std->begy;
-	refresh();
-	cbreak();
-	noecho();
-
-	while (1){
-		if (!endOfFile) {
-			for (x = 0; x < lines - 1; x++) {
-				for (y = 0; y < columns; y++) {
-					buff[y] = !endOfFile ? getc(fp) : ' ';
-
-					switch ((int)buff[y]) {
-						case EOF:
-							buff[y] = ' ';
-							endOfFile = true;
-							break;
-						case '\n':
-							/*	End of the line, filling the rest of buffer */
-							memset(buff + y, ' ', columns - y);
-							y = columns;
-							break;
-						case '\t':
-							/*	Perform tab instert	*/
-							memset(buff + y, ' ', TAB_SIZE);
-							y += TAB_SIZE - 1;
-							break;
+	lines = std->endy - std->begy;
+	endwin();
+
+	while (1) {
+		for (x = 0; x < lines - 1; x++) {
+
+			for (i = 0; i < columns; i++) {
```

Comment:
Does it actually matter? If string is shorter than `columns` then according `buff[i > columns]` are not used at all.